homeassistant/components/idasen_desk/__init__.py

```diff
@@ -15,62 +14,98 @@
     EVENT_HOMEASSISTANT_STOP,
     Platform,
 )
-from homeassistant.core import Event, HomeAssistant
-from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.core import Event, HomeAssistant, callback
 from homeassistant.helpers import device_registry as dr
 from homeassistant.helpers.device_registry import DeviceInfo
 from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
 
 from .const import DOMAIN
 
-PLATFORMS: list[Platform] = [Platform.COVER]
+PLATFORMS: list[Platform] = [Platform.COVER, Platform.SWITCH]
 
 _LOGGER = logging.getLogger(__name__)
 
 
+class IdasenDeskCoordinator(DataUpdateCoordinator):
+    """Class to manage updates for the Idasen Desk."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        logger: logging.Logger,
+        name: str,
+        address: str,
+    ) -> None:
+        """Init IdasenDeskCoordinator."""
+
+        super().__init__(hass, logger, name=name)
+        self._address = address
+        self._expected_connected = False
+
+        self.desk = Desk(self.async_set_updated_data)
+
+    async def connect(self) -> bool:
+        """Connect to desk."""
+        _LOGGER.debug("Trying to connect %s", self._address)
+        ble_device = bluetooth.async_ble_device_from_address(
+            self.hass, self._address, connectable=True
+        )
+        if ble_device is None:
+            return False
```

Comment:
I plan to add better support for the device disappearing and coming back in another PR.